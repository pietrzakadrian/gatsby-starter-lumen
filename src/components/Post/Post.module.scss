@import "src/assets/scss/variables";
@import "src/assets/scss/mixins";

.post {
  .content {
    @include margin-auto();
  }

  .footer {
    max-width: $layout-post-width;
    @include padding(0, 0);
    @include margin(0, 0);
  }

  .comments {
    max-width: $layout-post-width;
    padding: 0!important;
    @include margin-auto();
  }
}

@include breakpoint-md {
  .post {
    .footer {
      @include padding-equal(0);
    }
  }
}
