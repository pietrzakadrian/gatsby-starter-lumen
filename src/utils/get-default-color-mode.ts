const getDefaultColorMode = (): "dark" | "light" => {
  if (typeof window === "undefined") {
    return "dark";
  }

  const mql = window.matchMedia("(prefers-color-scheme: dark)");
  const hasMediaQueryPreference = typeof mql.matches !== "undefined";
  if (hasMediaQueryPreference) {
    return mql.matches ? "dark" : "light";
  }

  return "dark";
};

export default getDefaultColorMode;
