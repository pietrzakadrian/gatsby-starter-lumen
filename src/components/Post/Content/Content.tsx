import React from "react";

import * as styles from "./Content.module.scss";
import { Author } from "@/components/Post/Author";

interface Props {
  title: string;
  body: string;
  date: string;
  timeToRead: number;
}

const Content: React.FC<Props> = ({ body, title, date, timeToRead }: Props) => (
  <div className={styles.content}>
    <h1 className={styles.title}>{title}</h1>

    <Author date={date} timeToRead={timeToRead} />

    <div className={styles.body} dangerouslySetInnerHTML={{ __html: body }} />
  </div>
);

export default Content;
