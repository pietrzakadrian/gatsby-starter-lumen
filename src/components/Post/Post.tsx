import React from "react";
import * as styles from "./Post.module.scss";
import { Form as NewsletterForm } from "@/components/Newsletter/Form";
import { Related } from "@/components/Post/Related";
import { useSiteMetadata } from "@/hooks";
import type { Edge, Node } from "@/types";
import { Comments } from "./Comments";
import { Content } from "./Content";
import { Copyright } from "./Copyright";
import { Tags } from "./Tags";

interface Props {
  post: Node;
  related: Array<Edge>;
}

const Post: React.FC<Props> = ({ post, related }: Props) => {
  const { html, timeToRead } = post;
  const { tagSlugs } = post.fields;
  const { tags, title, date } = post.frontmatter;
  const { copyright } = useSiteMetadata();

  return (
    <div className={styles.post}>
      <div className={styles.content}>
        <Content
          body={html}
          title={title}
          date={date}
          timeToRead={timeToRead}
        />
      </div>

      <div>
        <Copyright copyright={copyright} />
      </div>

      <div className={styles.footer}>
        {tags && tagSlugs && <Tags tags={tags} tagSlugs={tagSlugs} />}
      </div>

      <div className={styles.comments}>
        <Comments />
      </div>

      <div>
        <h2>Newsletter</h2>
        <NewsletterForm />
      </div>

      {related.length > 0 && (
        <>
          <hr />

          <div>
            <h2>Related articles</h2>
            <Related related={related} />
          </div>
        </>
      )}
    </div>
  );
};

export default Post;
