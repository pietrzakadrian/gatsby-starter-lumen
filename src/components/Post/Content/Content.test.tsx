import React from "react";

import { StaticQuery, useStaticQuery } from "gatsby";

import { Content } from "@/components/Post/Content";
import * as mocks from "@/mocks";
import { testUtils } from "@/utils";

const mockedStaticQuery = StaticQuery as jest.Mock;
const mockedUseStaticQuery = useStaticQuery as jest.Mock;

describe("Content", () => {
  beforeEach(() => {
    mockedStaticQuery.mockImplementationOnce(({ render }) =>
      render(mocks.siteMetadata),
    );

    mockedUseStaticQuery.mockReturnValue(mocks.siteMetadata);
  });

  test("renders correctly", () => {
    const props = {
      title: mocks.markdownRemark.frontmatter.title,
      body: mocks.markdownRemark.html,
      date: mocks.markdownRemark.frontmatter.date,
      timeToRead: mocks.markdownRemark.frontmatter.timeToRead,
    };

    const tree = testUtils
      .createSnapshotsRenderer(<Content {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
