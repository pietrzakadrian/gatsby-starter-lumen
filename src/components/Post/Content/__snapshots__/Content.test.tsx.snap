// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders correctly 1`] = `
<div>
  <h1>
    Perfecting the Art of Perfection
  </h1>
  <div>
    <div>
      <div>
        <strong>
          John Doe
        </strong>
        <div>
          <p>
            September 1, 2016
          </p>
          <span>
            â€”
          </span>
          <p>
            5
             min read
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    dangerouslySetInnerHTML={
      {
        "__html": "<p>An Essay on Typography by Eric Gill takes the reader back to the year 1930. The year when a conflict between two worlds came to its term. The machines of the industrial world finally took over the handicrafts.</p>",
      }
    }
  />
</div>
`;
